### DISCUSSION TITLE
Is there a way to generate the colors programatically?

### DISCUSSION THREAD

**Discussion by mhmdjaw** [radix-ui/colors#56] [Q&A]
In other words, instead of manually going to the [custom colors](https://www.radix-ui.com/colors/custom) tool and copying colors from there, generate them dynamically sort of like this:

```ts
import { generateLightAccent } from '@radix-ui/colors'

const lightAccentBlue = generateLightAccent('#3D63DD')

lightAccentBlue.light4
lightAccentBlue.lightA8
```

This would be such an awesome feature and would enable dynamic theme generation. I would really love to work on this, but I can't find the source code for the [custom colors](https://www.radix-ui.com/colors/custom) tool.

**Comment by yan-yanishevsky** [comment]
Found in other repo:
https://github.com/radix-ui/website/blob/main/components/generateRadixColors.tsx
